client/src/Register.js

```diff
@@ -31,8 +33,9 @@ function Register() {
     event.preventDefault();
     setError(null);
     try {
-      await Api.auth.register(user);
-      history.push('/login', { flash: 'Your account has been created!' });
+      const response = await Api.auth.register(user);
+      authContext.setUser(response.data);
```

Comment:
Here we take the new user returned by the server after registration and set it in the Auth context, which makes it available throughout the site as the logged in user.